tests/common/test_command_populate_aip_checksums.py

```diff
@@ -0,0 +1,287 @@
+    package_uuid = uuid.uuid4()
+    package_current_path = f"uncompressedaip-{package_uuid}"
+    package_dir = pathlib.Path(aip_storage_fs_location.full_path) / package_current_path
+    package_dir.mkdir()
+
+    # Add tag manifest to fake a valid bag.
+    (package_dir / "tagmanifest-sha256.txt").touch()
+
+    result = models.Package.objects.create(
+        uuid=package_uuid,
+        status=models.Package.UPLOADED,
+        current_location=aip_storage_fs_location,
+        current_path=package_current_path,
+        package_type=models.Package.AIP,
+    )
+
+    assert not result.is_compressed
+
+    return result
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def aip_deleted_fs_location(aip_storage_fs_location: models.Location) -> models.Package:
+    return models.Package.objects.create(
+        package_type=models.Package.AIP,
+        status=models.Package.DELETED,
+        current_location=aip_storage_fs_location,
+        current_path="deleted.7z",
+    )
+
+
+@pytest.mark.django_db
+def test_command_fails_when_there_are_no_uploaded_aips(
+    aip_deleted_fs_location: models.Package,
+) -> None:
+    with pytest.raises(CommandError, match="No AIPs with status UPLOADED found"):
+        call_command("populate_aip_checksums")
+
+
+@pytest.mark.django_db
+@mock.patch(
+    "archivematica.storage_service.common.management.commands.StorageServiceCommand.error"
+)
+@mock.patch(
+    "archivematica.storage_service.common.utils.get_compressed_package_checksum"
+)
+def test_command_fails_when_checksum_is_missing_for_compressed_aip(
+    get_compressed_package_checksum: mock.Mock,
+    error: mock.Mock,
+    capsys: pytest.CaptureFixture[str],
+    compressed_package: models.Package,
+    aip_storage_fs_location: models.Location,
+) -> None:
+    get_compressed_package_checksum.return_value = (None, "sha256")
+
+    call_command(
+        "populate_aip_checksums",
+        "--location-uuid",
+        aip_storage_fs_location.uuid,
+    )
+
+    error.assert_called_once_with(
+        f"Unable to retrieve checksum information from pointer file for compressed AIP {compressed_package.uuid}"
+    )
+    captured = capsys.readouterr()
+    assert (
+        captured.out.strip("\n")
+        == "Complete. Checksums added for 0 of 1 identified AIPs. See output for details."
+    )
+
+
+@pytest.mark.django_db
+@mock.patch(
+    "archivematica.storage_service.common.utils.get_compressed_package_checksum"
+)
+def test_command_for_compressed_aip(
+    get_compressed_package_checksum: mock.Mock,
+    capsys: pytest.CaptureFixture[str],
+    compressed_package: models.Package,
+    aip_storage_fs_location: models.Location,
+) -> None:
+    compressed_package_checksum = (
+        "c2924159fcbbeadf8d7f3962b43ec1bf301e1b4f12dd28a8b89ec819f3714747"
+    )
+    algo = "sha256"
+    get_compressed_package_checksum.return_value = (
+        compressed_package_checksum,
+        algo,
+    )
```

Comment:
The more I see these `common.utils.get_compressed_package_checksum` mocks the more I think having an actual pointer file would be more appropriate.

`test_package` has a helper to copy existing pointer file fixtures https://github.com/artefactual/archivematica-storage-service/blob/db5b964f29a9a08da9869ce2bd5e3549c1610e20/tests/locations/test_package.py#L76

Or if you don't want to use a fixture file you could create one inline like https://github.com/artefactual/archivematica/blob/2b3c4ba787a341354fa0e0320a39674e199ab5a5/tests/dashboard/fpr/test_command_import_pronom_ids.py#L41-L55